### ISSUE TITLE
socket.on() error callback not working

### ISSUE THREAD

**Issue by DiedB** [athombv/apps-sdk-issues#80]
On calling the callback function passed to a socket using `Homey.emit()` from a custom pairing view with an error, the error is not correctly passed back to the pairing view.

The response from the emit endpoint is the following:
```
{"code":500,"error":"Login details incorrect","error_description":"An unkown error has occured [unknown_error]"}
```
The error I'm passing from my socket is inside the `error` key of the object. The error that the socket is receiving (in the first argument of the callback) is inside the `error_description` key of the object. There is no way of accessing the `error` key from the pairing view.

**Comment by WeeJeWel** [comment]
As it's a websocket, which serializes your data to JSON, an `Error` instance is not supported. I suggest you send a string as first argument, e.g. `callback( err.message || err.toString() );`.

**Comment by DiedB** [comment]
Sorry for being unclear, I am already passing a plain string as the first argument to the callback. This worked fine in SDK v1, but it stopped working in SDK v2 (presumably).

I am not sure if this is working like you intended it to work. Just tried passing an `Error` instance, and this time around it actually does work:
```{"code":500,"error":"Login details incorrect","error_description":"Login details incorrect"}```

Behaviour is different from SDK v1, but I can't find this being documented anywhere.

**Comment by WeeJeWel** [comment]
Hm this might indeed have changed. But passing a string as 1st argument does not work in Homey `>=v2.0.0`?

**Comment by DiedB** [comment]
Nope, you get the behaviour I described in the first post here. The string goes into the `error` field of the response and is never available to the pairing view (as the `error_description` field seems to be passed as argument).
